cli/src/cli_util.rs

```diff
@@ -2209,11 +2226,14 @@ fn resolve_default_command(
                     "Run `jj config set --user ui.default-command log` to disable this message."
                 )?;
             }
-            let default_command = config
-                .get_string("ui.default-command")
-                .unwrap_or_else(|_| "log".to_string());
+            let default_command = args.unwrap_or_else(|_| vec!["log".to_string()]);
+
             // Insert the default command directly after the path to the binary.
-            string_args.insert(1, default_command);
+            let orig_args = string_args.clone();
+            string_args.clear();
+            string_args.push(orig_args[0].clone());
+            string_args.extend(default_command);
+            string_args.extend(orig_args.into_iter().skip(1));
```

Comment:
nit: perhaps `splice()` can be used here.
https://doc.rust-lang.org/std/vec/struct.Vec.html#method.splice